gateway/src/http_entrypoints.rs

```diff
@@ -788,7 +788,7 @@ async fn sp_component_active_slot_set(
     let component = component_from_str(&component)?;
     let slot = body.into_inner().slot;
 
-    sp.set_component_active_slot(component, slot)
+    sp.set_component_active_slot(component, slot, false)
```

Comment:
@mkeeter and/or @jgallagher can you confirm that `false` is what we want here?

Reply:
Perfect; thank you

Reply:
uhhhhh

`false` means that we will not persistently select that boot slot.  It is a policy question about what the `sp_component_active_slot_set` _should_ be doing, and I'm not sure of the answer.

When updating similar APIs, I've added a `persist: bool` to the function signature, to force callers to make a choice about it; we may want to do the same to `sp_component_active_slot_set` here.

Reply:
> When updating similar APIs, I've added a persist: bool to the function signature, to force callers to make a choice about it; we may want to do the same to sp_component_active_slot_set here.

Yep, this is what I meant by extending `SpComponentFirmwareSlot`, which is coming from MGS's client here.

Reply:
Github strikes again – I had the page open before you commented, so didn't see it until my comment was already submitted!